ci/Dockerfile

```diff
@@ -0,0 +1,11 @@
+ARG PHP_VERSION
+FROM php:$PHP_VERSION
+WORKDIR /php-ast
+ADD *.c *.h *.php config.m4 config.w32 package.xml LICENSE README.md ./
+
+# Assume compilation will be the time consuming step.
+# Add tests after compiling so that it's faster to update tests and re-run them locally.
+RUN phpize && CFLAGS='-Wall -O2 -Wextra' ./configure && make -j2 && make install
```

Comment:
Why it is necessary to explicitly pass `-O2`?

Reply:
The optimization choice was arbitrary - I just chose something that would be similar to what end users would use

Reply:
What I meant was, isn't this what you get anyway because you're building against a release PHP build?

Reply:
It doesn't seem like it - `CFLAGS=... ./configure` seems like it's overriding the cflags, not adding to them when I look at the generated Makefile.

And I wanted to add the `-Wall -Wextra` to be able to see if a given change introduces compiler warnings (in the web ui), despite it not affecting the test results or compilation success

Reply:
Switched to EXTRA_CFLAGS - this seems to exist in the generated Makefile, even in PHP 7.0